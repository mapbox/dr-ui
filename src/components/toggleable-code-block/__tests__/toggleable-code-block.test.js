import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './toggleable-code-block-test-cases.js';

describe('toggleable-code-block', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.objc.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.objc;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.java.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.java;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
