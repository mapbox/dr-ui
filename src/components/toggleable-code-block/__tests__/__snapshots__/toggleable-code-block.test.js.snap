// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleable-code-block Java highlighting renders as expected 1`] = `
<div
  className="my24"
>
  <div
    className="flex-parent-mm flex-parent--space-between-main-mm mb6"
  >
    <div
      className="flex-child"
    >
      <div
        className="inline-block txt-bold mb6 color-gray-dark"
      >
        ThisIsAFile.extension
      </div>
      <div
        className="inline-block ml12-mm"
      >
        <a
          className="unprose link"
          href="https://github.com/mapbox/"
          title="View ThisIsAFile.extension on GitHub"
        >
          <svg
            className="events-none icon inline-block align-t"
            focusable="false"
            role="presentation"
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          >
            <use
              xlinkHref="#icon-github"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            />
          </svg>
           View on GitHub
        </a>
      </div>
    </div>
    <div
      className="flex-child"
    >
      <div
        className="relative "
      >
        <fieldset>
          <div
            aria-required={true}
            className="toggle-group round-full bg-blue py3 px3"
            data-test="test-java-filename-input"
            id="test-java-filename"
            role="radiogroup"
          >
            <label
              className="toggle-container "
            >
              <input
                autoFocus={false}
                checked={false}
                name="test-java-filename"
                onChange={[Function]}
                type="radio"
                value="javascript"
              />
              <div
                className="txt-s py3 toggle--white toggle--active-blue toggle"
              >
                JavaScript
              </div>
            </label>
            <label
              className="toggle-container "
            >
              <input
                autoFocus={false}
                checked={false}
                name="test-java-filename"
                onChange={[Function]}
                type="radio"
                value="swift"
              />
              <div
                className="txt-s py3 toggle--white toggle--active-blue toggle"
              >
                Swift
              </div>
            </label>
            <label
              className="toggle-container "
            >
              <input
                autoFocus={false}
                checked={false}
                name="test-java-filename"
                onChange={[Function]}
                type="radio"
                value="objectiveC"
              />
              <div
                className="txt-s py3 toggle--white toggle--active-blue toggle"
              >
                Objective-C
              </div>
            </label>
            <label
              className="toggle-container "
            >
              <input
                autoFocus={false}
                checked={true}
                name="test-java-filename"
                onChange={[Function]}
                type="radio"
                value="java"
              />
              <div
                className="txt-s py3 toggle--white toggle--active-blue toggle"
              >
                Java
              </div>
            </label>
            <label
              className="toggle-container "
            >
              <input
                autoFocus={false}
                checked={false}
                name="test-java-filename"
                onChange={[Function]}
                type="radio"
                value="kotlin"
              />
              <div
                className="txt-s py3 toggle--white toggle--active-blue toggle"
              >
                Kotlin
              </div>
            </label>
          </div>
        </fieldset>
        <div
          role="alert"
        />
      </div>
    </div>
  </div>
  <div
    className="relative round z0 scroll-styled scroll-auto"
    style={
      Object {
        "backgroundColor": "#f4f7fb",
        "maxHeight": 480,
      }
    }
  >
    <pre
      className="my-neg12 py12 ml-neg12 pl12 mobile-snippet"
    >
      <code
        className="px0 hljs"
      >
        <div
          className="relative z2"
          data-chunk-code="chunk-0"
        >
          <div
            className="pr12 opacity50 bg-darken10"
            style={
              Object {
                "paddingLeft": 12,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w36"
              data-content={1}
              style={
                Object {
                  "marginLeft": -12,
                  "marginTop": -0,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token keyword\\">public</span> <span class=\\"token keyword\\">class</span> <span class=\\"token class-name\\">MainActivity</span> <span class=\\"token keyword\\">extends</span> <span class=\\"token class-name\\">AppCompatActivity</span> <span class=\\"token punctuation\\">{</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
        </div>
        <div
          className="relative z2"
          data-chunk-code="chunk-1"
        >
          <div
            className="pr12 py3"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w30 py3"
              data-content={2}
              style={
                Object {
                  "marginLeft": -20.45,
                  "marginTop": -3,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token keyword\\">private</span> MapView mapView<span class=\\"token punctuation\\">;</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
        </div>
        <div
          className="relative z2"
          data-chunk-code="chunk-2"
        >
          <div
            className="pr12 opacity50 bg-darken10"
            style={
              Object {
                "paddingLeft": 12,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w36"
              data-content={3}
              style={
                Object {
                  "marginLeft": -12,
                  "marginTop": -0,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": " ",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
          <div
            className="pr12 opacity50 bg-darken10"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w36"
              data-content={4}
              style={
                Object {
                  "marginLeft": -26.45,
                  "marginTop": -0,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token annotation punctuation\\">@Override</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
        </div>
        <div
          className="relative z2"
          data-chunk-code="chunk-3"
        >
          <div
            className="pr12 py3"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w30 py3"
              data-content={5}
              style={
                Object {
                  "marginLeft": -20.45,
                  "marginTop": -3,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token keyword\\">protected</span> <span class=\\"token keyword\\">void</span> <span class=\\"token function\\">onCreate</span><span class=\\"token punctuation\\">(</span>Bundle savedInstanceState<span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
          <div
            className="pr12 py3"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w30 py3"
              data-content={6}
              style={
                Object {
                  "marginLeft": -20.45,
                  "marginTop": -3,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token keyword\\">super</span><span class=\\"token punctuation\\">.</span><span class=\\"token function\\">onCreate</span><span class=\\"token punctuation\\">(</span>savedInstanceState<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
          <div
            className="pr12 py3"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w30 py3"
              data-content={7}
              style={
                Object {
                  "marginLeft": -20.45,
                  "marginTop": -3,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Mapbox<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">getInstance</span><span class=\\"token punctuation\\">(</span><span class=\\"token keyword\\">this</span><span class=\\"token punctuation\\">,</span> <span class=\\"token string\\">\\"your-token\\"</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
          <div
            className="pr12 py3"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w30 py3"
              data-content={8}
              style={
                Object {
                  "marginLeft": -20.45,
                  "marginTop": -3,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token function\\">setContentView</span><span class=\\"token punctuation\\">(</span>R<span class=\\"token punctuation\\">.</span>layout<span class=\\"token punctuation\\">.</span>activity_main<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
        </div>
        <div
          className="relative z2"
          data-chunk-code="chunk-4"
        >
          <div
            className="pr12 opacity50 bg-darken10"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w36"
              data-content={9}
              style={
                Object {
                  "marginLeft": -26.45,
                  "marginTop": -0,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "mapView <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">(</span>MapView<span class=\\"token punctuation\\">)</span> <span class=\\"token function\\">findViewById</span><span class=\\"token punctuation\\">(</span>R<span class=\\"token punctuation\\">.</span>id<span class=\\"token punctuation\\">.</span>mapView<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
          <div
            className="pr12 opacity50 bg-darken10"
            style={
              Object {
                "paddingLeft": 26.45,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w36"
              data-content={10}
              style={
                Object {
                  "marginLeft": -26.45,
                  "marginTop": -0,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "mapView<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">onCreate</span><span class=\\"token punctuation\\">(</span>savedInstanceState<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
          <div
            className="pr12 opacity50 bg-darken10"
            style={
              Object {
                "paddingLeft": 12,
              }
            }
          >
            <div
              className="absolute fl color-gray align-r pr6 w36"
              data-content={11}
              style={
                Object {
                  "marginLeft": -12,
                  "marginTop": -0,
                }
              }
              unselectable="on"
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"token punctuation\\">}</span>",
                }
              }
              style={
                Object {
                  "marginLeft": 43.349999999999994,
                  "textIndent": -14.45,
                }
              }
            />
          </div>
        </div>
      </code>
    </pre>
    <div
      className="bg-lighten50 absolute left right"
      data-chunk-overlay="chunk-1"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <div
        className="bg-blue h-full w6"
      />
    </div>
    <div
      className="bg-lighten50 absolute left right"
      data-chunk-overlay="chunk-3"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <div
        className="bg-blue h-full w6"
      />
    </div>
    <div
      className="absolute z3 right mr3 color-white"
      data-chunk-copy="chunk-1"
      style={
        Object {
          "opacity": 0,
          "transition": "opacity 300ms linear",
        }
      }
    />
    <div
      className="absolute z3 right mr3 color-white"
      data-chunk-copy="chunk-3"
      style={
        Object {
          "opacity": 0,
          "transition": "opacity 300ms linear",
        }
      }
    />
  </div>
</div>
`;
